src/calculi/wlp.rs

```diff
@@ -0,0 +1,44 @@
+use std::fmt;
```

Comment:
I feel like these implementation files require surprisingly lots of boilerplate. Why not have one struct type for a calculus annotation that maintains the `Ident` and an enum `Calculus` (with values `Wp`, `Wlp`, etc.)? Then you have one (or multiple) functions that simply match on the `Calculus` value for the visitor.  I don't see the need to pass these abstract `dyn Trait` objects around if the set of possible calculi is always known by Caesar and not extensible.